### DISCUSSION TITLE
Input keeps values after removing

### DISCUSSION THREAD

**Discussion by zluizfs** [react-hook-form/react-hook-form#8434] [Q&A]
Hey Everyone!
I'm working with arrays and I'm having a problem when I'm going to delete an array it deletes it, but it doesn't update the input values it keeps the values.

Here we have two items that I added and I will remove the first one

![image](https://user-images.githubusercontent.com/49575536/171408285-79b38c0b-9093-4aea-bcf7-9a42fcdf2e7e.png)

I removed the first one but it kept the values of the first and deleted the last one. When I visualized the values using the JSON.stringify the values are correct it removed the first and kept the values of the second.

![image](https://user-images.githubusercontent.com/49575536/171408616-aba2a948-c15a-4ea8-8f2a-c740b9eae514.png)

But in the inputs it does not update the values when I remove.

Has anyone seen something like this and could help me?

I'm using the material ui and below is the code for a part of the form.

`const { fields, remove, append } = useFieldArray({
    control,
    name: `dadosPessoas.${person}.veiculoDados`
  })

    <>
      <Category title="Vechiles"
        noContainerBottom={true}
        noTitleBottom={fields.length > 0 ? true : false}
      />

      {fields.map((e, id) =>
        <div key={id}>
          <SubCategory
            title={id}
            handleChange={() => remove(id)}
          >

            <Controller
              name={`dadosPessoas.${person}.veiculoDados.${id}.marca`}
              defaultValue={e.marca}
              control={control}
              render={({ field }) => (
                <CustomTextField
                  defaultValue={e.marca}
                  label="Marca *"
                  value={field.value}
                  onChange={field.onChange}
                  inputRef={field.ref}
                  error={Boolean(errors.dadosPessoas?.[person].veiculoDados?.[id].marca)}
                  helperText={errors.dadosPessoas?.[person].veiculoDados?.[id].marca?.message}
                />
              )}
              rules={{ required: true }}
            />

          </SubCategory>
        </div>
      )}

      <ControlButton
        title="Add"
        disabled={fields.length >= 5}
        onClick={() => {
          clearErrors(`dadosPessoas.${person}.veiculoDados`)
          append({
            marca: '',
            modelo: '',
            ano: '',
            valor: ''
          })
        }}
      />
    </>`

**Comment by Moshyfawn** [comment]
A repro codesanbox instead would be greatly appreciated
